samples/snippets/src/main/java/com/example/bigquery/UpdateRoutine.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bigquery;
+
+// [START bigquery_update_routine]
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Routine;
+import com.google.cloud.bigquery.RoutineId;
+
+// Sample to update routine
+public class UpdateRoutine {
+
+  public static void runUpdateRoutine() {
+    // TODO(developer): Replace these variables before running the sample.
+    String datasetName = "MY_DATASET_NAME";
+    String routineName = "MY_ROUTINE_NAME";
+    updateRoutine(datasetName, routineName);
+  }
+
+  public static void updateRoutine(String datasetName, String routineName) {
+    try {
+      // Initialize client that will be used to send requests. This client only needs to be created
+      // once, and can be reused for multiple requests.
+      BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+
+      Routine routine = bigquery.getRoutine(RoutineId.of(datasetName, routineName));
+      routine.toBuilder().setBody("x * 4").build().update();
```

Comment:
could we setBody to something else? I don't really understand what the string here means... maybe something straightforward such as "new body text"?

Reply:
@stephaniewang526  retrieve the existing routine and updating the body value into a routine , you could check below reference code.

https://github.com/googleapis/java-bigquery/blob/ae97b5fec36daed60a4f51d1e5bdd3438aa079aa/samples/snippets/src/main/java/com/example/bigquery/CreateRoutine.java#L48-L59